servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/TaskBasedAsyncCompletableOperator.java

```diff
@@ -41,17 +42,30 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(TaskBasedAsyncCompletableOperator.class);
 
+    private volatile boolean hasOffloaded;
```

Comment:
this state is duplicated in `AbstractOffloadedSingleValueSubscriber#hasOffloaded`. I would suggest we keep the state at the lower layer it is harder to understand what threads access the state, and in practice I don't think it needs to be volatile.

For example the `SubscribeOn#handleSubscribe` usage of `offload` doesn't need to share state with the `Subscription` (it will happen before, ordering doesn't really matter).